### DISCUSSION TITLE
Not being able to import set_alpha() attribute

### DISCUSSION THREAD

**Discussion by Anirudh-Pranesh** [py-pdf/fpdf2#1269] [General]
Hi, I have a program that is using fpdf. I recently decided to include a watermark for my pdf and uninstalled fpdf and installed fpdf2. but despite doing this, I am not able to import the set_alpha() attribute and python gives the error : 

AttributeError: 'PDF' object has no attribute 'set_alpha'

Here is the code in question : 

```
from fpdf import FPDF
class PDF(FPDF):
    def watermark(self, image_path):
        """Add a watermark to the PDF."""
        self.image(image_path, x=10, y=10, w=190, h=190)  # Adjust size and position as needed
        self.set_alpha(0.1)  # Set transparency (0 to 1) --------------> Exception raised here
        self.image(image_path, x=10, y=10, w=190, h=190)  # Draw the watermark
        self.set_alpha(1)  # Reset transparency to normal
```

I have intsalled and unintsalled fpdf2 multiple times and nothing changes, it keeps giving the same error. For more context, the code was written by another colleague of mine, who used the same module fpdf2. It seems to be working for him but when i run this code on my system with the same module imported, the code raises an exception as decribed above. I am not sure about the python version and module version he used but here is mine : 

python version : 3.11.4
fpdf2 version : 2.7.9

Any help would be greatly appreciated ! Thanks.

**Comment by Lucas-C** [comment]
Hi @Anirudh-Pranesh

Where have you seen that `fpdf2` has a `set_alpha()` method? 🙂

The "alpha" channel is defined by your source image, that must support transparency (_.eg._ PNG and not JPG).

_cf._ https://py-pdf.github.io/fpdf2/Images.html#alpha-transparency

**Reply by Anirudh-Pranesh** [reply]
There's no method called set_alpha() ?? My colleauge insists the code works for him 🫤 ... Even asked some generative ai and it also says to import fpdf2 and use set_alpha()...

**Reply by Lucas-C** [reply]
No, sorry, there is no method called `set_alpha()` in `fpdf2`.

There is one in PyMuPDF, parhaps you mixed up libraries: https://pymupdf.readthedocs.io/en/latest/pixmap.html#Pixmap.set_alpha

**Reply by Anirudh-Pranesh** [reply]
Alright, thanks Lucas. Appreciate your help
